cdap-docs/developers-manual/source/building-blocks/transactional-messaging-system.rst

```diff
@@ -0,0 +1,287 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2017 Cask Data, Inc.
+    :description: Transactional Messaging System, a ACID-guaranteed "publish-and-subscribe" messaging service
+
+.. _transactional-messaging-system:
+
+==============================
+Transactional Messaging System
+==============================
+
+.. topic::  **Note: Beta Feature** 
+
+    TMS is currently a **beta** feature of CDAP |release|, and is subject to change without notice.
+
+
+Overview
+========
+The Transactional Messaging System (TMS) is a CDAP service that provides a
+"publish-and-subscribe" messaging system that understands transactions.
+
+It uses HBase for the persistent and durable storage of messages, and supports both
+transactional and non-transactional message publishing and consumption.
+
+TMS provides ACID *(atomicity, consistency, isolation,* and *durability)* guarantees,
+using `Apache Tephra <http://tephra.incubator.apache.org>`__ for the ACI portion. TMS
```

Comment:
No need to call out `ACI`. In fact the TMS itself has nothing to do with ACID (probably just the durable part). Probably better just not to have this paragraph.